ddtrace/debugging/_capture/tracing.py

```diff
@@ -62,3 +67,61 @@ def exit(self, retval, exc_info, duration):
 
     def line(self):
         raise NotImplementedError("Dynamic line spans are not supported in Python")
+
+
+@attr.s
+class SpanDecoration(CapturedEvent):
+    """Decorate a span."""
+
+    def _decorate_span(self, _locals):
+        # type: (t.Dict[str, t.Any]) -> None
+        probe = t.cast(SpanDecorationMixin, self.probe)
+
+        if probe.target_span == SpanDecorationTargetSpan.ACTIVE:
+            span = ddtrace.tracer.current_span()
+        elif probe.target_span == SpanDecorationTargetSpan.ROOT:
+            span = ddtrace.tracer.current_root_span()
+        else:
+            log.error("Invalid target span for span decoration: %s", probe.target_span)
+            return
+
+        if span is not None:
+            log.debug("Decorating span %r according to span decoration probe %r", span, probe)
+            for d in probe.decorations:
+                if not (d.when is None or d.when(_locals)):
+                    continue
+                for tag in d.tags:
+                    span.set_tag(tag.name, tag.value(_locals))
+
+    def enter(self):
+        # type: () -> None
+        probe = self.probe
+        if not isinstance(probe, SpanDecorationFunctionProbe):
+            log.debug("Span decoration entered with non-span decoration probe: %s", self.probe)
+            return
+
+        if probe.evaluate_at == ProbeEvaluateTimingForMethod.ENTER:
+            self._decorate_span(dict(self.args) if self.args else {})
+
+        self.state = CaptureState.DONE
+
+    def exit(self, retval, exc_info, duration):
+        # type: (t.Any, ExcInfoType, float) -> None
+        probe = self.probe
+
+        if not isinstance(probe, SpanDecorationFunctionProbe):
+            log.debug("Span decoration exited with non-span decoration probe: %s", self.probe)
+            return
+
+        if probe.evaluate_at == ProbeEvaluateTimingForMethod.EXIT:
+            self._decorate_span(self._enrich_args(retval, exc_info, duration))
+
+    def line(self):
+        probe = self.probe
+        if not isinstance(probe, SpanDecorationLineProbe):
+            log.debug("Span decoration on line with non-span decoration probe: %s", self.probe)
+            return
+
+        self._decorate_span(self.frame.f_locals)
+
+        self.state = CaptureState.DONE_AND_COMMIT
```

Comment:
why this is DONE_AND_COMMIT?

Reply:
This is what we do with line snapshot probes